QSB/PlayerInfo.cs

```diff
@@ -1,15 +1,54 @@
-﻿using UnityEngine;
+﻿using System.Linq;
+using OWML.Common;
+using QSB.Animation;
+using QSB.Utility;
+using UnityEngine;
 
 namespace QSB
 {
     public class PlayerInfo
     {
-        public uint NetId { get; set; }
+        public uint NetId { get; }
         public GameObject Body { get; set; }
         public GameObject Camera { get; set; }
+        public QSBFlashlight FlashLight => Camera.GetComponentInChildren<QSBFlashlight>();
         public string Name { get; set; }
-        public bool Ready { get; set; }
+        public bool IsReady { get; set; }
         public Transform ReferenceSector { get; set; }
         public State State { get; set; }
+
+        public QSBTool Signalscope
+        {
+            get
+            {
+                var tools = Camera.GetComponentsInChildren<QSBTool>();
+                if (tools.Length == 0)
+                {
+                    DebugLog.ToConsole("Error - Zero items in QSBTool list while trying to get Signalscope", MessageType.Error);
```

Comment:
if the debug line can be removed this whole thing can be shortened to one line:
`public QSBTool Signalscope => Camera.GetComponentsInChildren<QSBTool>().FirstOrDefault(x => x.Type == ToolType.Signalscope);`

Reply:
i simplified it after testing it